bin/git-cp

```diff
@@ -50,37 +50,29 @@ else
       fi
     fi
 
-    MERGE_OPT=
-    ff=$(git config --get merge.ff || true)
-    if [[ "$ff" == "only" ]]; then
-        MERGE_OPT="--ff"
-    fi
 
     echo "Copying $CURRENT_FILENAME into $DESTINATION_FILENAME"
 
-    INTERMEDIATE_FILENAME="${CURRENT_FILENAME//\//__}-move-to-${DESTINATION_FILENAME//\//__}"
-
-    # We keep the existing file on the side in a commit
-    git mv "${CURRENT_FILENAME}" "${INTERMEDIATE_FILENAME}"
-    git commit -nm "Keep $CURRENT_FILENAME"
+    # Pre-check that the source and destination will work
+    git mv --dry-run "${CURRENT_FILENAME}" "${DESTINATION_FILENAME}"
 
-    # We come back to the previous state and revert that change
-    INTERMEDIATE_SAVED=$(git rev-parse HEAD)
-    git reset --hard HEAD^
+    # Make a new branch and switch to it
+    BRANCH_NAME="git-cp-$(date +%s)"
+    git checkout -b "$BRANCH_NAME"
 
-    # We move the file to its new destination
+    # Move the original file to the new destination, in this branch
     git mv "${CURRENT_FILENAME}" "${DESTINATION_FILENAME}"
     git commit -nm "Copy $CURRENT_FILENAME into $DESTINATION_FILENAME"
 
-    # We come back to the previous state and revert that change again
-    DESTINATION_SAVED=$(git rev-parse HEAD)
-    git reset --hard HEAD^
+    # Restore the original file to keep it in the history
+    git checkout HEAD~ "${CURRENT_FILENAME}"
+    git commit -nm "Restore $CURRENT_FILENAME"
 
-    # We keep both files
-    # shellcheck disable=SC2086
-    git merge $MERGE_OPT "${DESTINATION_SAVED}" "${INTERMEDIATE_SAVED}" -m "Duplicate ${CURRENT_FILENAME} history."
+    # Switch to the original branch and merge this back in.
+    git checkout -
+    git merge --no-ff "$BRANCH_NAME" -m "Copying $CURRENT_FILENAME into $DESTINATION_FILENAME"
```

Comment:
This produces a commit with the repeated message (there is already a message `Copying ....` before it).
Could we optimize it?

Reply:
I'd be happy to change the commit messages to be more descriptive.  The reason I made them like this is because this merge message is the key one directly on the branch, so I figured it should be a summary message for the whole operation.

Here's an example of what results when I used this version of git-cp, run thrice, to turn one file into four copies--setting up for a commit that prunes down all four into just the parts each should keep:
 <img width="390" alt="image" src="https://github.com/user-attachments/assets/b0bdae6d-9a9e-479c-8d2a-13df69c6a8c1">

With this context, what do you think are the best messages?

Reply:
Do you prefer these messages?
<img width="362" alt="image" src="https://github.com/user-attachments/assets/6882e786-54d7-46ce-8dcf-ffa838231b89">

Reply:
These messages look good to me.

Reply:
Great!  I have pushed the better commit messages and taken this PR out of Draft.  Thanks, @spacewander .